OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -63,6 +63,8 @@ public class ArmamentInfo : ConditionalTraitInfo, Requires<AttackBaseInfo>
 		[Desc("Use multiple muzzle images if non-zero")]
 		public readonly int MuzzleSplitFacings = 0;
 
+		public readonly bool ShootAtClosestTargetablePosition = false;
```

Comment:
This would make more sense as an `enum TargetPosition { CenterPosition, Footprint }` enum that is defined on the `ProjectileInfo` or the `WeaponInfo` if that won't work.  This gives more readable yaml and stops it leaking into the ProjectileInfo.